custom/icds_reports/utils/aggregation_helpers/distributed/agg_dashboard_activity.py

```diff
@@ -0,0 +1,276 @@
+from datetime import timedelta
+from django.db.models import Max
+
+from custom.icds_reports.utils.aggregation_helpers.distributed.base import BaseICDSAggregationDistributedHelper
+from custom.icds_reports.const import  AGG_DASHBOARD_ACTIVITY
+from django.utils.functional import cached_property
+from django.contrib.auth.models import User
+from corehq.apps.users.dbaccessors.all_commcare_users import get_user_docs_by_username
+from dimagi.utils.chunked import chunked
+
+
+class DashboardActivityReportAggregate(BaseICDSAggregationDistributedHelper):
+    aggregate_parent_table = AGG_DASHBOARD_ACTIVITY
+
+    def __init__(self, date):
+        self.date = date
+        self.last_agg_date = self.get_last_agg_date()
+
+    def aggregate(self, cursor):
+        drop_query = self.drop_table_query()
+        create_table_query, create_table_param = self.create_table_query()
+        add_query, add_params = self.add_missing_users()
+
+        rollover_query, rollover_param = self.rollover_previous_data()
+        update_queries = self.update_queries()
+
+        cursor.execute(drop_query)
+        cursor.execute(create_table_query, create_table_param)
+        cursor.cursor.execute(add_query, add_params)
+        cursor.execute(rollover_query, rollover_param)
```

Comment:
why doesn't this rollover the previous data, and then add new users/update the old users times?

Reply:
That way I would also need to add another SQL query to delete the users from this tables which are "deleted" from the `User` table and looking into that, this was looking easier to me.
The basic flow is: insert latest available list of users then update their info from the last day activity table.